onnxruntime/core/providers/dnnl/subgraph/dnnl_maxpoolgrad.h

```diff
@@ -334,38 +376,36 @@ class DnnlMaxPoolGrad : public DnnlKernel {
   }
 
  private:
-  size_t src_size_;
-  size_t dst_size_;
-
-  std::shared_ptr<dnnl::memory> src_mem_;
   std::shared_ptr<dnnl::memory> diff_dst_mem_;
+  std::shared_ptr<dnnl::memory> diff_dst_mem_gpu_;
 
-  std::unique_ptr<dnnl::memory::desc> src_md_;
   std::unique_ptr<dnnl::memory::desc> diff_dst_md_;
 
   std::unique_ptr<dnnl::pooling_forward::desc> fwd_desc_;
-  std::shared_ptr<dnnl::pooling_forward::primitive_desc> fwd_primitive_desc_;
+
   std::shared_ptr<DnnlPool<T>> pool_fwd_;
 
   std::unique_ptr<dnnl::pooling_backward::desc> bwd_desc_;
   std::unique_ptr<dnnl::pooling_backward::primitive_desc> bwd_primitive_desc_;
   std::unique_ptr<dnnl::primitive> pool_bwd_;
 
-  std::shared_ptr<dnnl::memory> src_mem_from_;
-  std::unique_ptr<dnnl::memory> src_mem_to_;
-
   std::shared_ptr<dnnl::memory> diff_dst_mem_from_;
-  std::unique_ptr<dnnl::memory> diff_dst_mem_to_;
 
   std::unique_ptr<dnnl::memory> dst_mem_from_;
   std::unique_ptr<dnnl::memory> dst_mem_to_;
 
+  bool gpu_available_;
  private:
   dnnl::memory::format_tag GetAVXFormat(const dnnl::memory::dims& src_dims_mkl) {
     bool is_2D = src_dims_mkl.size() == 4 ? true : false;
     bool is_1D = src_dims_mkl.size() == 3 ? true : false;
```

Comment:
Not in this PR, but I should have caught this last time... the ternary operator here is redundant.

```c++
bool is_2D = src_dims_mkl.size() == 4;
bool is_1D = src_dims_mkl.size() == 3;
```

Reply:
I can update this.